benchmarks/benchmarks/common.py

```diff
@@ -10,12 +10,243 @@
+                raise SkipNotImplemented(f"{backend} skipped by @xp_capabilities")
+        else:
+            capabilities = {"jax_jit": False}
+
+        # Potentially wrap namespace with array_api_compat
+        xp = array_namespace(xp.empty(0))
+
+        self.xp = xp
+        self.func = func
+        self.warmup = False
+
+        if backend == "torch":
+            import torch
+
+            torch.set_default_dtype(torch.float64)
+            try:
+                torch.empty(0, device=device)
+            except RuntimeError:
+                raise SkipNotImplemented(f"{device=} not available")
+            torch.set_default_device(device)
+
+            if device == "cuda":
+                def wrapper(*args, **kwargs):
+                    res = func(*args, **kwargs)
+                    torch.cuda.synchronize()
+                    return res
+
+                self.func = wrapper
+
+        elif backend == "jax.numpy":
+            import jax
+
+            jax.config.update("jax_enable_x64", True)
+            try:
+                jax_device = jax.devices(device)[0]
+            except RuntimeError:
+                raise SkipNotImplemented(f"{device=} not available")
+            jax.config.update("jax_default_device", jax_device)
+
+            if capabilities["jax_jit"]:
+                func = jax.jit(func, static_argnames=static_argnames,
+                                static_argnums=static_argnums)
+                self.warmup = True
+
+            def wrapper(*args, **kwargs):
+                res = func(*args, **kwargs)
+                jax.block_until_ready(res)
+                return res
+
+            self.func = wrapper
+
+        elif backend == "dask.array":
+            import dask
+
+            def wrapper(*args, **kwargs):
+                res = func(*args, **kwargs)
+                return dask.compute(res)[0]
+
+            self.func = wrapper
+
+        elif backend == "cupy":
+            import cupy
+            # The default stream is non-blocking.
+            # As of CuPy 13.4.1, explicit non-blocking streams
+            # are substantially slower.
+            # cupy.cuda.Stream(non_blocking=True).use()
+
+            def wrapper(*args, **kwargs):
+                res = func(*args, **kwargs)
+                cupy.cuda.get_current_stream().synchronize()
+                return res
+
+            self.func = wrapper
+
+        else:
+            assert backend in ("numpy", "array_api_strict", "cupy")
+
+    def synchronize(self, *arrays):
+        """Wait until the given arrays have finished generating and return a
+        synchronized instance of them.
+        You need to call this on all arrays that your setup() function creates.
+        """
+        if self.backend == "dask.array":
+            import dask
+
+            arrays = dask.persist(*arrays)
+        elif self.backend in ("jax.numpy:cpu", "jax.numpy:cuda"):
+            import jax
+
+            jax.block_until_ready(arrays)
```

Comment:
Today this is unnecessary on jax:cpu, but it's not spelled out explicitly in the documentation and it's perfectly reasonable for JAX to kick off a worker thread in the future.